### ISSUE TITLE
Why do we need to apply mask while fine tuning?

### ISSUE THREAD

**Issue by pranoy-k** [huggingface/pytorch-openai-transformer-lm#53]
In attention class, you have the following code for masking. I understand the logic for pre training, but in fine tuning if we dont include language model loss we should have a check here for not applying the mask. Do we have to always apply the masking because the model was trained that way, is there an intuitive idea for this, because I dont see a necessity to do it experimentally 

This is the line I am talking about
w = w * self.b + -1e9 * (1 - self.b)  # TF implem method: mask_attn_weights

**Comment by ahug** [comment]
As you've already mentioned, one of the reason is that you combine the task-specific and the language model loss during fine-tuning.

According to my understanding, the mask should still be applied even if you don't include the LM loss since the model was trained that way, i.e. it learned to summarize the words it has seen so far (words on the left).

**Comment by pranoy-k** [comment]
That is very Helpful Andreas, thank you very much!
Do you know of anyway we can use the model for sequences of length more than 512, I have some medical documents which I want to fine-tune the model and they are typically of length 1000. 
Also just a thought. 
I know even all these pretrained models(including bert and open ai gpt) are using learned position embeddings instead of sinusoidal position embedding as proposed in original transformer paper, but is it just for empirical benefit?

**Comment by ahug** [comment]
Since the position embeddings are only learned for 512 positions, I can't think of a straightforward way to apply it to longer sequences.  A simple thing you could try would be to just split the sequence into chunks/half <512 and later combining the results/states.

In "Attention is all you need" the authors state that:
_"We also experimented with using learned positional embeddings instead and found that the two versions produced nearly identical results."_

Therefore I guess that they are learning the positional embeddings because they observed empirical gains on their tasks.

**Comment by pranoy-k** [comment]
Thanks Andreas, this are useful comments
